### DISCUSSION TITLE
Nested item becomes non-proxied after moving between nested lists

### DISCUSSION THREAD

**Discussion by notifybee** [Alfred-Skyblue/vue-draggable-plus#141] [Q&A]
Hi,

After dragging an item from one nested list to another, is it correct that the item becomes a non-proxied object?

In this instance I moved `card.1` from another board, and it changes to a non-proxied object.  

![image](https://github.com/Alfred-Skyblue/vue-draggable-plus/assets/13155555/e321cda0-964c-4c27-a08f-2d84f3c6e4f3)

The code for the nested boards is

```
                 <VueDraggable
                        @end="onEndBoard"
                         v-model=boardList class="flex flex-row min-h-64">
                            <div v-for="board in boardList" :key="board" class="w-80 rounded-md m-4 p-4 text-left bg-slate-100">
                                {{ board.name }}

                                <VueDraggable
                                @end="onEndCard"
                                v-model=board.card group="g1" class="min-h-full" :id="board.id">
                                    <div v-for="itemCard in board.card" :id="itemCard.id" :key="itemCard" class="w-80 bg-white rounded-md m-4 p-4 text-left bg-slate-100">
                                        {{ itemCard }}
                                    </div>
                                </VueDraggable>
                            </div>
                        </VueDraggable>
```

I ask because of this bug in inertiajs https://github.com/inertiajs/inertia/issues/775 and wondered if the non-proxied object was breaking it.

**Comment by Alfred-Skyblue** [comment]
Due to the internal use of JSON for deep copying, it will cause the loss of Proxy.

https://github.com/Alfred-Skyblue/vue-draggable-plus/blob/ef2cc8fec295a2d0006a68c8a0194defd777cf40/src/useDraggable.ts#L30-L33

 You can pass a clone function that returns the proxy object:
 
 ```html
<script setup>
import { reactive } from 'vue'

function clone(el) {
  const val = JSON.parse(JSON.stringify(el))
  return reactive(val)
}
</script>

<template>
  <VueDraggable :clone="clone">
    <!-- code   -->
  </VueDraggable>
</template>

                           
 ```
